proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,806 @@
+
+## Current Behavior
+
+The default `dotnet build` behavior in .NET Core 3.0 Preview 4 is the following:
+
+* Framework-dependent application
+* Architecture-specific executable (matches architecture of the SDK )
+* Native dependencies for all achitectures
+
+This combination of characteristics, particularly the last two points, isn't entirely coherent. It gives you an executable for one architecture, but native dependencies for all achitectures. That means that the application will run on all architecture supported by the native dependencies (if any) but that an executable is provided for only one architecture and the `dotnet` launcher needs to be used on all other architectures.
+
+It also means that the application may be carrying extra files it doesn't need if the application is intended to be run on only one architecture. We expect this to be the case for WPF and Windows Forms applications, for example.
+
+As a result, the current behavior is considered to be unfortunate. It doesn't seem like the right behavior for developers intentionally targeting multiple architectures or for targeting single architectures.
+
+An example of the current default experience follows, with an application that has a single NuGet dependency (from Microsoft):
+
+```console
+C:\git\testapps\cpumath>type cpumath.csproj
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.ML.CpuMath" Version="0.11.0" />
+  </ItemGroup>
+
+</Project>
+
+C:\git\testapps\cpumath>dotnet build
+
+C:\git\testapps\cpumath>dir /s /b bin\*.dll bin\*.exe bin\*.so bin\*.dylib
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\cpumath.dll
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\Microsoft.ML.CpuMath.dll
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\cpumath.exe
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\linux-x64\native\libCpuMathNative.so
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\osx-x64\native\libCpuMathNative.dylib
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\win-x64\native\CpuMathNative.dll
+C:\git\testapps\cpumath\bin\Debug\netcoreapp3.0\runtimes\win-x86\native\CpuMathNative.dll
+
+C:\git\testapps\cpumath>bin\Debug\netcoreapp3.0\cpumath.exe
+Hello World!
+```
+
+Notice the inclusion of native dependencies for Linux, macOS and two Windows architectures. At the same time, the application includes an executable that will only run on Windows x64. If the application is launched with `dotnet cpumath.dll`, it will then run on all of the architectures supported by the native dependencies.
+
+## Proposed Behavior
+
+`dotnet build` should have three modes:
+
+* Architecture-agnostic, framework-dependent, application that includes native dependencies for all supported architectures and no EXE
```

Comment:
Are there any development scenarios where missing the EXE would be a detriment?

Reply:
Yes. Having an EXE is nice for launching an application from explorer, for example, even for web servers.